catalog/spatial/geocoding/spatial-clavin-lucene/src/main/java/org/codice/ddf/spatial/clavin/ClavinFacadeImpl.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+
+package org.codice.ddf.spatial.clavin;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.bericotech.clavin.ClavinException;
+import com.bericotech.clavin.GeoParser;
+import com.bericotech.clavin.GeoParserFactory;
+import com.bericotech.clavin.index.IndexDirectoryBuilder;
+import com.bericotech.clavin.resolver.ResolvedLocation;
+
+public class ClavinFacadeImpl implements ClavinFacade {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClavinFacadeImpl.class);
+
+    private String clavinIndexLocation;
+
+    private GeoParser clavinParser;
+
+    public void setClavinIndexLocation(final String indexLocation) {
+        this.clavinIndexLocation = indexLocation;
+    }
+
+    @Override
+    public List<ResolvedLocation> getResolvedLocations(String document)
+            throws ClavinFacadeException {
+        try {
+            return clavinParser.parse(document);
+        } catch (Exception e) {
+            // clavin Geoparser#parse throws Exception !
+            throw new ClavinFacadeException("Clavin failed to resolve geo location. ", e);
+        }
+    }
+
+    @Override
+    public void createIndex(String resource) throws ClavinFacadeException {
```

Comment:
Does this ever get called when the system is restarted?  It seems you only create the calvinParser when the command line interface selects a file to index.